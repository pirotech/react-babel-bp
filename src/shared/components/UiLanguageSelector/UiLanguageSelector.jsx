import React from 'react';
import { withTranslation } from 'react-i18next';
import './UiLanguageSelector.scss';

class UiLanguageSelector extends React.Component {
  constructor() {
    super();
  }

  onChangeLanguage = () => {
    const {i18n} = this.props;
    i18n.changeLanguage(i18n.language === 'en' ? 'ru' : 'en');
  };

  render() {
    const {i18n} = this.props;

    return (
      <div className="ui-language-selector">
        <button onClick={this.onChangeLanguage}>{i18n.language}</button>
      </div>
    );
  }
}

export default withTranslation('common')(UiLanguageSelector);