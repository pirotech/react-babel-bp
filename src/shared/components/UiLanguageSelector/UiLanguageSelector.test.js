import React from 'react';
import UiLanguageSelector from './UiLanguageSelector';
import { mount, shallow, render } from 'enzyme';

describe('Language selector test', () => {
  it('Snapshot test with mount', () => {
    const wrapper = mount(<UiLanguageSelector />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Snapshot test with shallow', () => {
    const wrapper = shallow(<UiLanguageSelector />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Snapshot test with render', () => {
    const wrapper = render(<UiLanguageSelector />);
    expect(wrapper).toMatchSnapshot();
  });
});
