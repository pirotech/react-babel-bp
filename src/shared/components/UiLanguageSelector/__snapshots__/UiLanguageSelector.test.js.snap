// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language selector test Snapshot test with mount 1`] = `
<withI18nextTranslation(UiLanguageSelector)>
  <UiLanguageSelector
    i18n={Object {}}
    t={[Function]}
    tReady={false}
  >
    <div
      className="ui-language-selector"
    >
      <button
        onClick={[Function]}
      />
    </div>
  </UiLanguageSelector>
</withI18nextTranslation(UiLanguageSelector)>
`;

exports[`Language selector test Snapshot test with render 1`] = `
<div
  class="ui-language-selector"
>
  <button />
</div>
`;

exports[`Language selector test Snapshot test with shallow 1`] = `
<UiLanguageSelector
  i18n={Object {}}
  t={[Function]}
  tReady={false}
/>
`;
